src/token/libexec/token/token-permit

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+### token-permit -- sign (and submit) EIP2612 or DAI style permits
+set -e
+display_usage() {
+  echo "Usage: token permit <token> <spender> [<amount|allowed>] [<deadline|expiry>] [--submit]"
+}
+
+for i in "$@"; do
+  case $i in
+    --submit)
+      SUBMIT=yes
+      shift
+      ;;
+    *)
+      shift
+      set -- "$@" "$i"
+      ;;
+  esac
+done
+
+if [ $# -lt 2 ];  then
+    display_usage
+    exit 1
+fi
+
+TOKEN=$1
+SPENDER=$2
+
+EIP2612_TYPEHASH=$(seth keccak $(seth --from-ascii "Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)"))
+PERMIT_TYPEHASH=$(seth call $TOKEN "PERMIT_TYPEHASH()(bytes32)")
+
+if [ "$EIP2612_TYPEHASH" != "$PERMIT_TYPEHASH" ]; then
+    DAI=yes
```

Comment:
Are there other tokens that use `dai` style permits? It might be more robust to hardcode the `dai` address here?

Reply:
Chai uses it.